### DISCUSSION TITLE
How to get into configure private mode on Juniper

### DISCUSSION THREAD

**Discussion by sanjay49erfan** [ktbyers/netmiko#2212] [Q&A]
Hello Kirk,

I have a Python script using Netmiko that I use for pushing config changes to Juniper MX240 edge routers. I want to configure it using " configure private" mode rather then the default mode. And reason is that there might be some other engineer working on some part of the config on that router that was not completed and not committed. When I run my script, it does a commit at the configure (default) mode and throws an error since the engineer did not complete his configuration. How can I get into configure private mode?

This is how to get into configure private mode via CLI on Juniper MX240

```
--------------------------------------------------------------------------------
user@labedgemx240> configure ?
Possible completions:
  <[Enter]>            Execute this command
  batch                Work in batch mode (commit happens in batch)
  dynamic              Work in dynamic database
  exclusive            Obtain exclusive lock (other users cannot make changes)
  private              Work in private database (other's changes do not show)
  |                    Pipe through a command
```

part of my script
----------------------
```python
config_commands = ['set system syslog host 10.7.255.38 any any']
output = net_connect.send_config_set(config_commands)

output = net_connect.send_command('show configuration | grep 10.7.255.38')
print (output) 

output = net_connect.commit()
print(output)
```

thanks

**Comment by ktbyers** [comment]
`send_config_set` has a `config_mode` argument where you can specify the command to use to enter configuration mode. I would look at that.


Regards, Kirk
